src/commands/build.rb

```diff
@@ -108,7 +109,9 @@ def render_build_files(parameters, node)
       end
 
       def start_build
-        nodes.each(&:start_build)
+        nodes.each do |node|
+          Utils::MetalThread.new { node.start_build }
```

Comment:
I'm not sure we need the `MetalThread` class to manage the build threads - possibly we can use [ThreadGroup](http://devdocs.io/ruby~2.4/threadgroup) and achieve the same thing in a more standardized way? E.g. would the following do the same thing (untested):

```ruby

def build_threads
  @build_threads ||= ThreadGroup.new
end

def start_build
  nodes.each do |node|
    build_threads.add(Thread.new { node.start_build })
  end
end

def clear_up_build_threads
  build_threads.list.map(&:kill)
end

def teardown
  clear_up_build_threads
  clear_up_built_node_marker_files
  Output.stderr 'Done.'
end
```

Reply:
This has been implemented by rebasing the the old commits. https://github.com/alces-software/metalware/pull/181/commits/2345c4d2cfcfbe116eb62fdf5cb5fbd37bff63e7